scu_builders.py

```diff
@@ -0,0 +1,335 @@
+"""Functions used to generate scu build source files during build time
+"""
+import glob, os
+import math
+from pathlib import Path
+from os.path import normpath, basename
+
+base_folder = os.path.abspath("./") + "/"
+_verbose = False
+_is_release_build = False
+
+
+def clear_out_existing_files(output_folder, extension):
+    output_folder = os.path.abspath(output_folder)
+    # print("clear_out_existing_files from folder: " + output_folder)
+
+    if not os.path.isdir(output_folder):
+        # folder does not exist or has not been created yet,
+        # no files to clearout. (this is not an error)
+        return
+
+    os.chdir(output_folder)
```

Comment:
Can we avoid changing dir ? this would create bug hard to catch if for some reason we forget to switch back to the correct location ?

I'd also suggest using the `pathlib.Path` object which is much nicer to use than the lower level `os.path` stuff

```python
def clear_out_existing_files(output_folder: Path, extension: str):
    if not output_folder.is_dir():
        return

    for file in output_folder.glob(f".{extension}"):
        file.unlink()
```